### PR TITLE
docs: update onboarding readme

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Updates the Google Services section and yarn section of the onboarding README.

## **Related issues**

Fixes: new developer onboarding issues related to google-services configs

## **Manual testing steps**

1. Try using the readme to onboard as a "new developer"

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,44 +39,36 @@ git clone git@github.com:MetaMask/metamask-mobile.git && \
 cd metamask-mobile
 ```
 
-### **Firebase Messaging Setup**
+#### Firebase Messaging Setup
```
Comment (Prithpal-Sooriya) [inline README.md:42]: These are subheading under "Building the app". Happy to take other recommendations on doc structuring.

Larger restructures lets do as a separate PR.

File: README.md [hunk 2]
```diff
@@ -39,44 +39,36 @@ git clone git@github.com:MetaMask/metamask-mobile.git && \
...
 MetaMask uses Firebase Cloud Messaging (FCM) to enable app communications. To integrate FCM, you’ll need configuration files for both iOS and Android platforms.
 
-#### **Configuration Files Required**
-- **`GoogleService-Info.plist`** (iOS)
-- **`google-services.json`** (Android)
+##### Internal Contributor instructions
 
-These files are essential for FCM integration and are automatically generated when running: `yarn start:ios` or `yarn start:android`
+1. Grab the `.js.env` file from 1Password, ask around for the correct vault. This file contains the `GOOGLE_SERVICES_B64_ANDROID` and `GOOGLE_SERVICES_B64_IOS` secrets that will be used to generate the relevant configuration files for IOS/Android.
+2. [Install](./README.md#install-dependencies) and [run & start](./README.md#running-the-app) the application as documented below.
```
Comment (Prithpal-Sooriya) [inline README.md:49]: IMPORTANT - this should simplify the flow for our MM developers.

The previous docs were very confusing, which caused developers running additional commands when they did not need to!

File: README.md [hunk 3]
```diff
@@ -85,7 +77,7 @@ export GOOGLE_SERVICES_B64_IOS="$(base64 -w0 -i ./ios/GoogleServices/GoogleServi
 In case of any doubt, please follow the instructions in the link below to get your Firebase project config file.
 [Firebase Project Quickstart](https://firebaseopensource.com/projects/firebase/quickstart-js/messaging/readme/#getting_started)
 
-**Install dependencies**
+#### Install dependencies
```
Comment (Prithpal-Sooriya) [inline README.md:80]: Installing dependencies should be under the "Building the app" header/section.

Again, if we want larger discussions and rearchitecting of documentation, lets leave it to a different PR.

File: docs/readme/environment.md [hunk 4]
```diff
@@ -25,6 +25,18 @@ It is recommended to install a Ruby version manager such as [rbenv](https://gith
...
+  <summary>Troubleshooting</summary>
+
+  ```bash
+    # Inspect that ruby is in path
+    which ruby
+
+    # Ensure you are using the correct ruby version
+    ruby --version
+  ```
+</details>
```
Comment (Prithpal-Sooriya) [inline docs/readme/environment.md:38]: To test the issue developers were having, I re-onboarded this repo after removing ruby and acting as a "new user".

This step was not that informative, and I've added troubleshooting steps to help ensure devs check their ruby version.

File: docs/readme/environment.md [hunk 5]
```diff
@@ -55,11 +67,29 @@ Install node version defined in the file `.nvmrc`
 
 ### Yarn v1
 
-With the correct Node version installed, Yarn v1 can be installed sudo-less in your system using `npm`
+Ensure you are using the correct yarn version (yarn v1) as noted in the `package.json`.
```
Comment (Prithpal-Sooriya) [inline docs/readme/environment.md:70]: We have a `packageManager` property inside our [package.json](https://github.com/MetaMask/metamask-mobile/blob/ba7c2cc99b790d9d8490c5dde7cb6964269967ac/package.json#L580)

I think we should recommend using corepack instead of npm. Thoughts?

**Review by Daniel-Cross — APPROVED** [review]

**Comment by Prithpal-Sooriya** [comment]
I will resolve comments and add to merge queue tomorrow if no additional comments are made.

**Review by gambinish — APPROVED** [review]

**Review by mathieuartu — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Prithpal-Sooriya) [reply README.md:42]: no comments, so resolving.

File: docs/readme/environment.md (hunk 4 above)
Comment (Prithpal-Sooriya) [reply docs/readme/environment.md:38]: no comments, so resolving. There should be no issue with this change.

File: docs/readme/environment.md (hunk 5 above)
Comment (Prithpal-Sooriya) [reply docs/readme/environment.md:70]: No comments, resolving. This should be no issue, and we should encourage the use of corepack. Especially if we move to yarnV2, or a different version, corepack will still be a valid command (unlike npm install which only installs v1)
